src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -1254,6 +1254,11 @@ public void SendFile(string? fileName, byte[]? preBuffer, byte[]? postBuffer, Tr
         /// <exception cref="SocketException">An error occurred when attempting to access the socket.</exception>
         public void SendFile(string? fileName, ReadOnlySpan<byte> preBuffer, ReadOnlySpan<byte> postBuffer, TransmitFileOptions flags)
         {
+            if (_protocolType == ProtocolType.Udp)
+            {
+                throw new NotSupportedException(SR.net_notconnected);
+            }
```

Comment:
The async variant checks for `if (!IsConnectionOriented)` while the sync version checks `(!Connected)`. I think that - instead of limiting the error case explicitly to UDP - we should consolidate these checks, throwing in case of `(!IsConnectionOriented || !Connected)`.